gapic/src/main/com/google/gapid/views/AtomTree.java

```diff
@@ -450,7 +455,13 @@ public void updateElement(Object parent, int index) {
       AtomStream.Node child = ((AtomStream.Node)parent).getChild(index);
       atoms.load(child, refresher::refresh);
       viewer.replace(parent, index, child);
-      viewer.setChildCount(child, child.getChildCount());
+      if (child.getChildCount() > 100) {
+        // For nodes with lots of children, only allocate the children, once the node
+        // is actually expanded.
+        viewer.setHasChildren(child, true);
```

Comment:
Oh, I see. Yes we could. The only benefit of giving the actual counts here is that when a node is expanded, the UI will already knows how much to grow the view and the initial UI update will be a bit quicker. I'm fine with simplifying this, if you'd like, as it doesn't make a big difference.